bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/clangd/internal/ui/ClangFormatConfigurationPage.java

```diff
@@ -46,6 +47,9 @@ public void setContainer(IPreferencePageContainer container) {
 		if (workspace == null) {
 			workspace = PlatformUI.getWorkbench().getService(IWorkspace.class);
 		}
+		if (formatFile == null) {
+			formatFile = PlatformUI.getWorkbench().getService(ClangFormatFile.class);
```

Comment:
something is wrong with this sanity initialization, perhaps we can do it once in constructor

Reply:
yes, getting the service via the workbench in init fails. I'll remove the init method.

Reply:
...because init won't be called.